IndomitableIbises/Bangazon-Sprint-3
# Add Category Form

## Link to Ticket
References #8 and #26 

## Description of Proposed Changes
- User can view all categories, click add category, fill in the form, submit, and see the updated category list
- Updated faker_factory with Category Model change

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout RC-CategoryForm
```
1. **Shut down** your server and **close** DB Browser
1. Run faker_factory script: `./django_data.sh website faker_factory` or `django_data.sh website faker_factory`
1. **Open DB Browser** to ensure all appropriate tables and data exist
1. Run Server: `python manage.py runserver`
1. **Register** a new user
1. Once logged in, go to this url: `http://127.0.0.1:8000/categories`
1. Click **Add Category** button
1. Enter a new category and press enter
1. You will be redirected the a list of categories where you will see your newly created category.
1. Shake yer rump



## Impacted Areas in Application

List general components of the application that this PR will affect:

* add_category.py
* list_categories.py
* add_category.html
* list_category.html
* faker_factory.py


## Mentions @username

Tag users that need to review this code
@sirwin4 @paulzimmclay @eagobert 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
